src/main/java/com/linkedin/kmf/apps/MultiClusterMonitor.java

```diff
@@ -70,28 +72,31 @@ public MultiClusterMonitor(Map<String, Object> props, String name) throws Except
 
   @Override
   public void start() {
-    _topicManagementService.start();
-    _produceService.start();
-    _consumeService.start();
-    LOG.info(_name + "/MultiClusterMonitor started");
+    _multiClusterTopicManagementService.start();
+    CompletableFuture<Void> completableFuture = _multiClusterTopicManagementService.topicManagementReady();
+    completableFuture.thenRun(() -> {
+      _produceService.start();
+      _consumeService.start();
+    });
+    LOG.info(_name + "/MultiClusterMonitor started.");
   }
 
   @Override
   public void stop() {
-    _topicManagementService.stop();
+    _multiClusterTopicManagementService.stop();
     _produceService.stop();
     _consumeService.stop();
     LOG.info(_name + "/MultiClusterMonitor stopped");
   }
 
   @Override
   public boolean isRunning() {
-    return _topicManagementService.isRunning() && _produceService.isRunning() && _consumeService.isRunning();
+    return _multiClusterTopicManagementService.isRunning() && _produceService.isRunning() && _consumeService.isRunning();
   }
 
   @Override
   public void awaitShutdown() {
```

Comment:
I know that it is out of the scope of this PR. But I think changing the `awaitShutdown()` method in the `App` interface and make it take some timeout parameter might be a better practice. For example, refer to the `ExecutorService` API `boolean awaitTermination(long timeout, TimeUnit unit)`. No need to change now. But it would be great if you can consider this change to be on the backlog.

Reply:
This could be in my backlog collections for revisions in the future.